test/unitTests/logging/WarningMessageObserver.test.ts

```diff
@@ -77,39 +84,91 @@ suite('WarningMessageObserver', () => {
         getOmnisharpServerOnErrorEvent("someText", "someFile", 1, 2)
     ].forEach((event: BaseEvent) => {
         suite(`${event.constructor.name}`, () => {
+
             test(`When the event is fired then a warning message is displayed`, () => {
-                observer.post(event);
-                scheduler.advanceBy(1500); //since the debounce time is 1500 no output should be there
-                expect(warningMessage).to.be.equal("Some projects have trouble loading. Please review the output for more details.");
+                let marble = `${timeToMarble(1500)}a`;
+                let marble_event_map = { a: event };
+                let eventList = scheduler.createHotObservable(marble, marble_event_map);
+                eventList.subscribe(e => observer.post(e));
+                scheduler.expectObservable(assertionObservable).toBe(`${timeToMarble(3000)}a`, { a: '[1] Some projects have trouble loading. Please review the output for more details.' });
+                scheduler.flush();
+
+                expect(warningMessages.length).to.be.equal(1);
+                expect(warningMessages[0]).to.be.equal("Some projects have trouble loading. Please review the output for more details.");
             });
 
             test(`When events are fired rapidly, then they are debounced by 1500 ms`, () => {
-                observer.post(event);
-                scheduler.advanceBy(1000);
-                expect(warningMessage).to.be.undefined;
-                observer.post(event);
-                scheduler.advanceBy(500);
-                expect(warningMessage).to.be.undefined;
-                scheduler.advanceBy(1000);
-                expect(warningMessage).to.not.be.empty;
-                //once there is a silence for 1500 ms the function will be invoked
+                let marble = `${timeToMarble(1000)}a${timeToMarble(500)}b${timeToMarble(500)}c`;
+
+                let eventB = getOmnisharpMSBuildProjectDiagnosticsEvent("BFile",
+                    [getMSBuildDiagnosticsMessage("warningFile", "", "", 1, 2, 3, 4)],
+                    [getMSBuildDiagnosticsMessage("errorFile", "", "", 5, 6, 7, 8)]);
+
+                let eventC = getOmnisharpMSBuildProjectDiagnosticsEvent("CFile",
+                    [getMSBuildDiagnosticsMessage("warningFile", "", "", 1, 2, 3, 4)],
+                    [getMSBuildDiagnosticsMessage("errorFile", "", "", 5, 6, 7, 8)]);
+
+                let marble_event_map = { a: event, b: eventB, c: eventC };
+                let eventList = scheduler.createHotObservable(marble, marble_event_map);
+                eventList.subscribe(e => observer.post(e));
+                scheduler.expectObservable(assertionObservable).toBe(`${timeToMarble(3520)}a`, { a: '[1] Some projects have trouble loading. Please review the output for more details.' });
```

Comment:
We can also check here that in an intermediate time say 2000, there is no output because of the debouncing

Reply:
And the event that will be emitted here will be event c, because of the debouncing event a and b will not be emitted and since c will have an emission silence it will be emitted